tiled/src/main/kotlin/ktx/tiled/mapObject.kt

```diff
@@ -0,0 +1,73 @@
+package ktx.tiled
+
+import com.badlogic.gdx.maps.MapObject
+import com.badlogic.gdx.maps.MapProperties
+import com.badlogic.gdx.maps.objects.*
+import com.badlogic.gdx.math.*
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapObject]. The type is automatically
+ * derived from the type of the given default value. If the property is not defined the defaultValue will be returned.
+ * @param key property name
+ * @param defaultValue default value in case the property is missing
+ * @return value of the property or defaultValue if property is missing
+ */
+inline fun <reified T> MapObject.property(key: String, defaultValue: T): T = this.properties[key, defaultValue, T::class.java]
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapObject]. If the property
+ * is not defined then this method returns null.
+ * @param key property name
+ * @return value of the property or null if the property is missing
+ */
+inline fun <reified T> MapObject.propertyOrNull(key: String): T? = this.properties[key, T::class.java]
+
+/**
+ * Extension method to directly access the [MapProperties] of a [MapObject] and its [containsKey][MapProperties.containsKey] method
+ * @param key property name
+ * @return true if the property exists. Otherwise false
+ */
+fun MapObject.containsProperty(key: String) = properties.containsKey(key)
+
+/**
+ * Extension property to retrieve the x-coordinate of the [MapObject]. If the object does not have
+ * a x-coordinate then 0 is returned
+ */
+val MapObject.x: Float
+    get() = property("x", 0f)
```

Comment:
Is this property always defined? An unexpected 0 without any errors or warnings could be misleading. I'd consider converting it to `propertyOrNull("x") ?: throw MissingPropertyException("x")`.

Now that I think about it, `property(name: String)` extension method that return a non-null value or throws an exception would also be a useful utility.

Reply:
From what I can tell from my maps at least and from the `com.badlogic.gdx.maps.tiled.BaseTmxMapLoader#loadObject` there are some default properties for every object.
Those are `x`, `y`, `width`, `height` and `id` (I did not know about width and height. Just noticed that now and will add them).

There would be other properties like `rotation` and `type` but they can be `null` and are therefore not added and I would not suggest an extension for them due to that.

That's why those properties should never be missing and the default value will never apply. 

I can add the exception extension if you want but personally - at least in my use cases - I prefer a default value instead of an exception.
E.g. I prefer to have an enemy created at 0,0 instead of not creating him at all. The first scenario gives me the hint that everything is working but something is wrong with the location where he is added. The second scenario could be anything since the enemy does not get created at all.
Also, in Kotlin there are no checked exceptions, correct? So a user would need to pro-actively add the try-catch block because the IDE will not warn him?

Again, I am openminded here. Just let me know what do you think is the better approach. Exception or default value - I will do as you wish ;)

Reply:
Exceptions are easier to track and patch. With enemies randomly created at `0, 0`, you wouldn't be sure if the source of the bug is your spawning code or the map settings. With an exception thrown during map loading, you immediately know that the file is corrupted.

It's true that there are no checked exceptions, but then again - if the map files are correct, these kind of errors are not expected at all. You give the user the choice to handle it as they prefer.

Reply:
>  There would be other properties like rotation and type but they can be null and are therefore not added and I would not suggest an extension for them due to that.

You can add nullable properties for these.

Reply:
changed the standard properties extensions as well to the version where it throws an exception instead of returning a default value